cmd/influxd/inspect/verify_seriesfile.go

```diff
@@ -9,10 +9,86 @@ import (
 	"sort"
 	"sync"
 
+	"github.com/influxdata/influxdb/v2/logger"
 	"github.com/influxdata/influxdb/v2/tsdb"
+	"github.com/spf13/cobra"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
+type args struct {
+	dir        string
+	db         string
+	seriesFile string
+	verbose    bool
+	concurrent int
+}
+
+func NewVerifySeriesfileCommand() *cobra.Command {
+	var arguments args
+	cmd := &cobra.Command{
+		Use:   "verify-seriesfile",
+		Short: "Verifies the integrity of series files.",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SetOut(os.Stdout)
+
+			config := logger.NewConfig()
+			config.Level = zapcore.WarnLevel
+			if arguments.verbose {
+				config.Level = zapcore.InfoLevel
+			}
+			log, err := config.New(os.Stderr)
+			if err != nil {
+				return err
+			}
+
+			v := NewVerify()
+			v.Logger = log
+			v.Concurrent = arguments.concurrent
+
+			if arguments.seriesFile != "" {
+				_, err := v.VerifySeriesFile(arguments.seriesFile)
+				return err
+			}
+
+			if arguments.db != "" {
+				_, err := v.VerifySeriesFile(filepath.Join(arguments.dir, arguments.db, "_series"))
+				return err
+			}
+
+			dbs, err := ioutil.ReadDir(arguments.dir)
+			if err != nil {
+				return err
+			}
+
+			for _, db := range dbs {
```

Comment:
We could DRY up this logic using something like this:
```go
var dbs []string
if arguments.seriesFile != "" {
    dbs = []string{arguments.seriesFile}
} else if arguments.db != "" {
    dbs = []string{filepath.Join(arguments.dir, arguments.db, "_series")}
} else {
    var err error
    dbs, err = ioutil.ReadDir(arguments.dir)
    # Handle err
}
for _, db := range dbs {
    # Existing code
}
```

Reply:
I was able to clean it up a bit, but not fully, since `dbs` in the last case is type `[]os.FileInfo` and not string, and it is important to keep it that way to check `Isdir()`